profiling/src/lib.rs

```diff
@@ -675,6 +686,7 @@ unsafe extern "C" fn minfo(module_ptr: *mut zend::ModuleEntry) {
             },
         );
 
+        #[cfg(feature = "allocation_profiling")]
```

Comment:
Let's print out a message saying the feature was not compiled in, since it's on by default and we've shipped it. Let's not do it for timeline since it's not ready.